### PR TITLE
fix: add frontend plugins for guessing platform

### LINKED ISSUE #3300: npx amplify-app "Cannot find module 'amplify-frontend-javascript'"
**Which Category is your question related to?**

Amplify CLI, DataStore, React

**Amplify CLI Version** 

4.13.1

**What AWS Services are you utilizing?**

Amplify: CLI, DataStore

**Provide additional details e.g. code snippets**

To reproduce:

```bash
$ npx create-react-app test --use-npm
$ cd test
$ npx amplify-app
```

Output from last command:

```
$ npx amplify-app
npx: installed 94 in 3.502s
✅ Found Amplify CLI version 4.13.1
🎸 Creating base Amplify project
Error: Cannot find module 'amplify-frontend-javascript'
Require stack:
- /home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/src/index.js
- /home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/bin/amplify-app
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:17)
    at Function.Module._load (internal/modules/cjs/loader.js:859:27)
    at Module.require (internal/modules/cjs/loader.js:1028:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at /home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/src/index.js:185:31
    at Array.forEach (<anonymous>)
    at guessPlatform (/home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/src/index.js:184:20)
    at getProjectConfig (/home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/src/index.js:151:51)
    at ChildProcess.<anonymous> (/home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/src/index.js:115:18)
    at ChildProcess.emit (events.js:305:20) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/src/index.js',
    '/home/ubuntu/.npm/_npx/1844/lib/node_modules/amplify-app/bin/amplify-app'
  ]
}
```

Seeing the same error on Ubuntu and macOS with node 12.x and 13.x .

Am I missing a step or depenency, or is this a bug? I don't remember hitting this with the latest version in December.

### PR DESCRIPTION
**Pull request by nikhname**
fix #3300

*Issue #, if available:*
#3300 
*Description of changes:*
Adds dependencies to guess platform in amplify-app and updates min version of amplify-app to 4.14.0
Adds E2E test for amplify-app (all platforms)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by yuth** [comment]
# [Codecov](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=h1) Report
> Merging [#3312](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=desc) into [master](https://codecov.io/gh/aws-amplify/amplify-cli/commit/6795e783c104004a2b2576f6903b35c1c6d2ed03&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master    #3312   +/-   ##
=======================================
  Coverage   60.51%   60.51%           
=======================================
  Files         254      254           
  Lines       12625    12625           
  Branches     2647     2648    +1     
=======================================
  Hits         7640     7640           
+ Misses       4646     4645    -1     
- Partials      339      340    +1     
```


| [Impacted Files](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...ges/amplify-graphql-types-generator/src/loading.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS1ncmFwaHFsLXR5cGVzLWdlbmVyYXRvci9zcmMvbG9hZGluZy50cw==) | `74.50% <0.00%> (ø)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=footer). Last update [6795e78...64c29c4](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3312?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by nikhname** [comment]
> Please E2E tests for amplify-app to make sure we're covering grounds.

Added e2e tests for javascript - will add e2e for android and ios in separate PRs

File: packages/amplify-app/src/test-helpers/amplify-app-cleanup.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        if (!err) {
+          resolve();
+        } else {
+          reject(err);
+        }
+      });
+  });
+}
+
+function deleteAmplifyAppFiles() {
```
Comment (attilah) [inline packages/amplify-app/src/test-helpers/amplify-app-cleanup.js]: can't we delete a full folder here in 1 step? rimraf support globbing as well.

Comment (nikhname) [reply packages/amplify-app/src/test-helpers/amplify-app-cleanup.js]: Yeah this is unnecessary with test folders being removed in the test suites. Removed

**Comment by andrestone** [comment]
Why so much protocol / bureaucracy for an obvious AND critical fix? The package is broken, meaning that everyone who finds DataStore an interesting idea and tried the Getting Started steps **for the past two weeks** suffered a huge knockback.

Am I missing something?

**Pull request merged**